lib/spack/spack/solver/asp.py

```diff
@@ -1878,12 +1879,12 @@ def external_spec_selected(self, pkg, idx):
         )
 
     def depends_on(self, pkg, dep, type):
-        dependency = self._specs[pkg]._dependencies.get(dep)
-        if not dependency:
-            self._specs[pkg]._add_dependency(
-                self._specs[dep], (type,))
+        dependencies = self._specs[pkg].edges_to_dependencies(name=dep)
```

Comment:
For now, to make this clearer, I'd suggest:

```suggestion
        dependencies = self._specs[pkg].edges_to_dependencies(name=dep)
        assert len(dependencies) == 1, "Current solver does not handle multiple dependency edges of the same name"
        dependency = dependencies[0] if dependencies else None
```

Reply:
Done in 5d77b88